pkg/hubble/cell/cell.go

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Hubble
+
+package hubblecell
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/hive/job"
+	"github.com/sirupsen/logrus"
+
+	"github.com/cilium/cilium/api/v1/models"
+	"github.com/cilium/cilium/pkg/cgroups/manager"
+	"github.com/cilium/cilium/pkg/endpointmanager"
+	identitycell "github.com/cilium/cilium/pkg/identity/cache/cell"
+	"github.com/cilium/cilium/pkg/ipcache"
+	k8sClient "github.com/cilium/cilium/pkg/k8s/client"
+	"github.com/cilium/cilium/pkg/k8s/watchers"
+	monitorAgent "github.com/cilium/cilium/pkg/monitor/agent"
+	"github.com/cilium/cilium/pkg/node"
+	nodeManager "github.com/cilium/cilium/pkg/node/manager"
+	"github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/pkg/recorder"
+	"github.com/cilium/cilium/pkg/service"
+)
+
+// The top-level Hubble cell, implements several Hubble subsystems: reports pod
+// network drops to k8s, Hubble flows based prometheus metrics, flows logging
+// and export, and a couple of local and tcp gRPC servers.
+var Cell = cell.Module(
+	"hubble",
+	"Exposes the Observer gRPC API and Hubble metrics",
+
+	cell.Provide(newHubbleIntegration),
+	cell.Config(defaultConfig),
+)
+
+type hubbleParams struct {
+	cell.In
+
+	IdentityAllocator identitycell.CachingIdentityAllocator
+	EndpointManager   endpointmanager.EndpointManager
+	IPCache           *ipcache.IPCache
+	ServiceManager    service.ServiceManager
+	CGroupManager     manager.CGroupManager
+	Clientset         k8sClient.Clientset
+	K8sWatcher        *watchers.K8sWatcher
+	NodeManager       nodeManager.NodeManager
+	NodeLocalStore    *node.LocalNodeStore
+	MonitorAgent      monitorAgent.Agent
+	Recorder          *recorder.Recorder
+
+	// NOTE: we still need DaemonConfig for the shared EnableRecorder flag.
+	AgentConfig *option.DaemonConfig
+	Config      config
+
+	// TODO: replace by slog
+	Logger logrus.FieldLogger
+}
+
+type HubbleIntegration interface {
+	Status(ctx context.Context) *models.HubbleStatus
+}
+
+func newHubbleIntegration(jobGroup job.Group, params hubbleParams) (HubbleIntegration, error) {
```

Comment:
non-blocking: add `job.Group` to `hubbleParams` like all the other dependencies

Reply:
Done